controllers/cruisecontroltask_controller.go

```diff
@@ -302,6 +348,69 @@ func (r *CruiseControlTaskReconciler) createCCOperation(
 	}, nil
 }
 
+// brokersJBODSelector filters out the JBOD and not JBOD brokers from a broker list based on the capacityConfig
+func brokersJBODSelector(brokerIDs []string, capacityConfigJSON string) (brokersJBOD []string, brokersNotJBOD []string, err error) {
+	// JBOD is generated by default
+	if capacityConfigJSON == "" {
+		return brokerIDs, nil, nil
+	}
+	brokerIsJBOD := make(map[string]bool)
+	for _, brokerID := range brokerIDs {
+		brokerIsJBOD[brokerID] = true
+	}
+
+	var capacityConfig koperatorccconf.JBODInvariantCapacityConfig
+	err = json.Unmarshal([]byte(capacityConfigJSON), &capacityConfig)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "could not unmarshal the user-provided broker capacity config")
+	}
+	for _, brokerCapacity := range capacityConfig.Capacities {
+		brokerCapacityMap, ok := brokerCapacity.(map[string]interface{})
+		if !ok {
+			continue
+		}
```

Comment:
Are we sure that just continuing here won't cause any issues? Should we at least log something?

Reply:
Here we use this: https://github.com/banzaicloud/koperator/blob/db78e506f76d23f6425f4a00867ef995f288ad6d/pkg/resources/cruisecontrol/configmap.go#L162 which comes from the past. When there is a syntax error in the kafkaCluster.Spec.CruiseControlConfig.CapacityConfig and we don't do any validation the Cruise Control will do and in the Cruise Control pod there will be error message and the pod will not be ready state. Because of that, we will not do any CC operation. That is why is safe to use here continue I think.